src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -85,7 +85,7 @@ public static Scope getScope(DetailAST ast) {
      */
     public static Scope getScopeFromMods(DetailAST aMods) {
         return Optional.ofNullable(getDeclaredScopeFromMods(aMods))
-                .orElseGet(() -> getDefaultScope(aMods.getParent()));
```

Comment:
We need an actual analysis of why this change is OK.

Reply:
done, please check the PR description